### ISSUE TITLE
Allow Skill Event to be entered by Submission

### ISSUE THREAD

**Issue by adamayd** [NationalSecurityAgency/skills-service#152]
**Is your feature request related to a problem? Please describe.**
I would like if the skills could be entered by a user entering a type of submission.

**Describe the solution you'd like**
Instead of clicking Report Skill, there would be an area to submit what the user did to complete the skill and even the ability to attach proof (picture/document) of completing the skill.  There would have to be an area in the dashboard that allows for higher level or admin users to approve or deny the submissions.

**Describe alternatives you've considered**
We've used the BadgeOS plugin for WordPress, for this same functionality, but it lacks the other capabilities that SkillTree provides as well as having to run a full WordPress instance for it to function.

**Additional context**
Is there something like this feature planned for the project already?  If not, could we have a conversation about the estimated lift to create this by collaboration or contribution.

**Comment by sudo-may** [comment]
@adamayd it’s great to hear from you!

This sounds like a very interesting idea! We will discuss it internally and get back to you sometime next week. Thank you very much!

**Comment by sudo-may** [comment]
@adamayd 

We discussed it internally and here is what we are thinking: 

Currently users of integrated applications can ONLY view their skills progress and ranking within that integrated application using SkillTree’s [Skills Display Component](https://code.nsa.gov/skills-docs/skills-client/vuejs.html#skills-display). Admins of an application can use the SkillTree dashboard to view users skill progress and ranking but not the users themselves. We currently have on our roadmap to add the ability for users to view their ranking and skill progress directly in the dashboard which will probably work well with what you are requesting and is probably a dependency to your request. 

With this new capability, non-admin users can then go directly to the Dashboard and view their skills and progress. One idea is to add a self-reporting button – something like “I did it” or “Done” on specific skills; It could also accept an explanation and as you mentioned maybe even screen shots. In this scenario a project admin would be able to enable self-reporting for all the skills or only select skills. Project admins can then also chose whether self-reported skills would have to go through the approval workflow OR just use the honor system. 

This is a complicated feature that would touch multiple facets of SkillTree architecture so we are thinking it would make sense to develop it internally. We added this feature to our roadmap projected to be worked on Summer of 2021. However, we very much welcome contributions, the best way is to pick up an unassigned ticket tagged with ['Good First Issue'](https://github.com/NationalSecurityAgency/skills-service/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22+no%3Aassignee)

Thanks again for such a wonderful idea!!

**Comment by adamayd** [comment]
@sudo-may 

This is great news.  I'll take this information back to my team and see what / how we could contribute.  I greatly appreciate you considering this and putting it on the roadmap.  I look forward to working with you in any capacity we can!!

**Comment by sudo-may** [comment]
@adamayd fyi - I am picking up this ticket now - we will target it for the next release (1.4.0)

**Comment by sudo-may** [comment]
PR: https://github.com/NationalSecurityAgency/skills-service/pull/327

**Comment by rmmayo** [comment]
![image](https://user-images.githubusercontent.com/47030042/108714459-8e623400-74e7-11eb-9cc9-ab096267858f.png)


@sudo-may 


a couple of things I noticed:  

- "Submitted Successfully" is in the middle of the other two sentences - this should probably be first, with the other two following it (maybe on a separate line)?
- Also, maybe reword the other sentences to "This skill requires approval from a project administrator.  Now let's play the waiting game!"
- and the last thing, the approval button, when you go to click it and are hovering over it, the background color it changes to is too dark and you cannot read the text in the button anymore.. we may need to invert the text color I think

**Comment by rmmayo** [comment]
merged PR: #356

**Comment by sudo-may** [comment]
@adamayd just circling back to let you know that this feature is now available as a part of [skills-service:1.4.0](https://github.com/NationalSecurityAgency/skills-service/releases/tag/1.4.0) release (current latest is [1.4.3](https://github.com/NationalSecurityAgency/skills-service/releases/tag/1.4.3) however). You can also take a look at the [Self Reporting](https://code.nsa.gov/skills-docs/dashboard/user-guide/self-reporting.html) sections of the docs. Feedback and comments are welcome. Have a wonderful day!

**Comment by adamayd** [comment]
@sudo-may Thanks for circling back.  I am no longer on the project that was looking at your solution, but I will surely pass this information along.
